### PR TITLE
Cleanups vmspawn

### PR DESCRIPTION
**Pull request by keszybz**
Post-review cleanups for #31218.

### CODE REVIEW

File: src/journal/journald-socket.c [hunk 1]
```diff
@@ -125,10 +114,8 @@ void server_forward_socket(
                         c = memchr(i->iov_base, '=', i->iov_len);
 
                         /* this should never happen */
-                        if (_unlikely_(!c || c == i->iov_base)) {
-                                log_error("Found invalid journal field, refusing to forward.");
-                                return;
-                        }
+                        if (_unlikely_(!c || c == i->iov_base))
+                                return log_error("Found invalid journal field, refusing to forward.");
```
Comment (YHNdnzj) [inline src/journal/journald-socket.c:118]: `_errno(SYNTHETIC_ERRNO(EINVAL), ...)`?

**Review by YHNdnzj — COMMENTED** [review]
If you touch the error handling, maybe change things so the error is propagated but ignored at caller's side? But doesn't matter too much

**Comment by keszybz** [comment]
> If you touch the error handling, maybe change things so the error is propagated but ignored at caller's side? But doesn't matter too much

I think it's fine as is. I just wanted to get rid of some of the verbose boilerplate.

**Comment by poettering** [comment]
> If you touch the error handling, maybe change things so the error is propagated but ignored at caller's side? But doesn't matter too much

i'd agree with @YHNdnzj. it's nicer to ignore at caller side. but also doesn't really matter i guess.

**Comment by keszybz** [comment]
OK, I reworked one of the functions to return int, and that caused some follow-up changes, so PTAL. (Only the first commit is changed). I didn't change the other function, because it'd require adding a few SYNTHETHIC_ERRNOs which seem worse than the one `(void)` needed now.

File: src/journal/journald-socket.h [hunk 2]
```diff
@@ -4,5 +4,4 @@
 #include "journald-server.h"
 #include "socket-util.h"
 
-void server_forward_socket(Server *s, const struct iovec *iovec, size_t n, int priority);
-void server_open_forward_socket(Server *s);
+int server_forward_socket(Server *s, const struct iovec *iovec, size_t n, int priority);
```
Comment (YHNdnzj) [inline src/journal/journald-socket.h:7]: Cast return value to void on caller's side?

File: src/journal/journald-socket.c [hunk 3]
```diff
@@ -13,39 +13,38 @@
...
+        if (connect(socket_fd, &addr->sockaddr.sa, addr->size) < 0)
+                return log_debug_errno(errno, "Failed to connect to remote address for forwarding, ignoring: %m");
 
         s->forward_socket_fd = TAKE_FD(socket_fd);
-        log_debug("Successfully connected to remote address for forwarding");
+        log_debug("Successfully connected to remote address for forwarding.");
+        return 1;
 }
 
 static inline bool must_serialise(struct iovec iov) {
```
Comment (YHNdnzj) [inline src/journal/journald-socket.c]: While at it, maybe -> `serialize` for consistency with the rest of the codebase?

File: src/journal/journald-socket.c [hunk 4]
```diff
@@ -58,61 +57,54 @@ static inline bool must_serialise(struct iovec iov) {
...
 
         return false;
 }
 
-void server_forward_socket(
+int server_forward_socket(
                 Server *s,
                 const struct iovec *iovec,
                 size_t n_iovec,
                 int priority) {
```
Comment (YHNdnzj) [inline src/journal/journald-socket.c:73]: (insert a newline here?)

File: src/journal/journald-socket.c [hunk 5]
```diff
@@ -58,61 +57,54 @@ static inline bool must_serialise(struct iovec iov) {
...
         _cleanup_free_ struct iovec *iov_alloc = NULL;
         struct iovec *iov = NULL;
-
         _cleanup_free_ le64_t *len_alloc = NULL;
         le64_t *len = NULL;
```
Comment (YHNdnzj) [inline src/journal/journald-socket.c]: `iov` and `len` doesn't need to be explicitly initialized BTW

**Review by YHNdnzj — APPROVED** [review]

**Comment by keszybz** [comment]
Updated with all the suggested changes.

**Pull request merged**